meilisearch-release-changelog/v1.0.0.md

```diff
@@ -0,0 +1,205 @@
+* The `--log-level` option and `MEILI_LOG_LEVEL` environment variable now only accept the following arguments: `ERROR`, `WARN`, `INFO`, `DEBUG`, `TRACE` and `OFF` (#3293) @loiclec
+* Remove hidden but usable CLI arguments `--nb-max-chunks` and `--log-every-n` (#3305) @Kerollmops
+
+## Revamp primary key inference
+
+We have improved the primary key inference process when receiving documents for the first time.
+
+If an index's primary key has not been specified, Meilisearch will attempt to infer the primary key by searching for any attributes ending with `id`, such as `puid` or` _id`.
+
+If only one such attribute is detected, it will be designated as the index's primary key.
+
+If multiple attributes ending with `id` are detected, Meilisearch will ask you to specify the primary key instead of choosing the first found one.
+
+If no primary key candidates are detected, Meilisearch will throw an error as before.
+
+Done in meilisearch/milli#752 and #3269 by @dureuill
+
+## Error handler changes
+
+When Meilisearch returns an error, a `code` field is returned in the JSON object. Multiples of these unclear and vague `code`s have been changed to improve the user experience.
+
+We consider this as a breaking change since, as a developer, it can lead to break your code if you catch one of the following `code`s in your code base:
+* When calling `POST /indexes/swap-indexes`
+  - `duplicate_index_found` is renamed into `invalid_swap_duplicate_index_found`
+  - Sending an array of indexes not containing exactly 2 `indexUids` for a swap operation object now returns `invalid_swap_indexes` instead of a `bad_request`
+* When calling `GET /indexes` and `GET /indexes/:uid`
+  - Using the `limit` parameter wrongly, Meilisearch now returns `invalid_index_limit` instead of `bad_request`
+  - Using the `offset` parameter wrongly, Meilisearch now returns `invalid_index_offset` instead of `bad_request`
+* When calling `POST /indexes` and `PUT /indexes/:uid`
+  - When `uid` is missing in the payload, Meilisearch returns `missing_index_uid` instead of `bad_request`
+  - Sending a value with a different type than `string` or `null` for `primaryKey` now returns `invalid_index_primary_key` instead of `bad_request`
+* When calling `GET /index/:uid/documents` and `GET /indexes/:uid/documents/:uid`
+  - Using the `fields` parameter wrongly, Meilisearch now returns `invalid_document_fields` instead of `bad_request`
+  - Using the `limit` parameter wrongly, Meilisearch now returns `invalid_document_limit` instead of `bad_request`
+  - Using the `offset` parameter wrongly, Meilisearch now returns `invalid_document_offset` instead of `bad_request`
+* When calling `POST /index/:uid/documents` and `PUT /indexes/:uid/documents`
+  - Using the `primaryKey` parameter wrongly, Meilisearch now returns ` invalid_index_primary_key` instead of `bad_request`
```

Comment:
```suggestion
  - Incorrect use of the `fields` parameter now returns `invalid_document_fields` instead of `bad_request`
  - Incorrect use of the `limit` parameter now returns `invalid_document_limit` instead of `bad_request`
  - Incorrect use of the `offset` parameter now returns `invalid_document_offset` instead of `bad_request`
* When calling `POST /index/:uid/documents` and `PUT /indexes/:uid/documents`
  - Incorrect use of the `primaryKey` parameter now returns ` invalid_index_primary_key` instead of `bad_request`
```